### DISCUSSION TITLE
Error: Cannot read property 'map' of undefined

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#292] [Capstones!] [answered]
Name: Nathan Keith

Describe your issue: When trying to read, I keep getting this error- 
_TypeError: Cannot read property 'map' of undefined._
<img width="530" alt="Screen Shot 2021-06-12 at 10 45 22 AM" src="https://user-images.githubusercontent.com/76710923/121781627-51d84800-cb6b-11eb-87ac-439bc5ffd09a.png">

What I have Googled:
https://stackoverflow.com/questions/24706267/error-cannot-read-property-map-of-undefined
https://www.debuggr.io/react-map-of-undefined/
https://www.akashmittal.com/cannot-read-property-map-undefined/

What have you tried? Tell us everything you have tried.
- I tried to give the variable an initial value with useState([])
- I'm in the process of reviewing all related calls and props
Link to branch in GitHub: https://github.com/nashvegasnate/SetList/tree/nk-read

**Comment by drteresavasquez** [comment, marked as answer]
@nashvegasnate is working to put the answer

**Comment by nashvegasnate** [comment]
So...all it took was a question mark. That easy. What happened was that my code was calling on an object property that doesn't yet exist, throwing the error noted above. By inserting the ? between the property name and .map, it will overlook the error and simply return as undefined instead. It is called optional chaining. I remember us going over it in class, but now I know when to use it and exactly what it is called! Now I can hopefully start reading...
